### ISSUE TITLE
Feature request: JSON output

### ISSUE THREAD

**Issue by dreamorosi** [webdeveric/validate-package-exports#1]
Thank you for creating this package!

I wanted to ask if you’d be open to add an option to the CLI that allows users to get the output in JSON format.

When using this mode, the output would become something like this:

```json
[
  {
    "type": "error",
    "file": "@aws-lambda-powertools/jmespath/package.json",
    "path": "./envelopes.require"
  }
]
```

Instead of this:

```sh

❌ require: @aws-lambda-powertools/jmespath/envelopes cannot be required (["exports","./envelopes","require"])
Error: Command failed: node --input-type=commonjs --eval="require('@aws-lambda-powertools/jmespath/envelopes');"
/workspaces/tmp-test/node_modules/@aws-lambda-powertools/jmespath/lib/cjs/envelopes.d.ts:1
import type { ParsingOptions, JSONObject } from './types.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at [eval]:1:1
    at runScriptInThisContext (node:internal/vm:144:10)

Node.js v20.11.1

    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: `node --input-type=commonjs --eval="require('@aws-lambda-powertools/jmespath/envelopes');"`,
  stdout: '',
  stderr: '/workspaces/tmp-test/node_modules/@aws-lambda-powertools/jmespath/lib/cjs/envelopes.d.ts:1\n' +
    "import type { ParsingOptions, JSONObject } from './types.js';\n" +
    '^^^^^^\n' +
    '\n' +
    'SyntaxError: Cannot use import statement outside a module\n' +
    '    at internalCompileFunction (node:internal/vm:77:18)\n' +
    '    at wrapSafe (node:internal/modules/cjs/loader:1288:20)\n' +
    '    at Module._compile (node:internal/modules/cjs/loader:1340:27)\n' +
    '    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)\n' +
    '    at Module.load (node:internal/modules/cjs/loader:1207:32)\n' +
    '    at Module._load (node:internal/modules/cjs/loader:1023:12)\n' +
    '    at Module.require (node:internal/modules/cjs/loader:1235:19)\n' +
    '    at require (node:internal/modules/helpers:176:18)\n' +
    '    at [eval]:1:1\n' +
    '    at runScriptInThisContext (node:internal/vm:144:10)\n' +
    '\n' +
    'Node.js v20.11.1\n'
}
```

This way users would be able to consume the output programmatically and incorporate it as part of their CI workflows.

Note that the suggested JSON format is just a proposal, as long as it’s a valid JSON it’d work for us.

**Comment by webdeveric** [comment]
@dreamorosi Sure. I can add that. How verbose do you need it? Seeing the serialized error is quite messy as it is now. Do you want the same data in the JSON?

**Comment by dreamorosi** [comment]
The most important parts would be to have:
- the path to the `package.json` file that contains the issue
- The path within the JSON of the faulty export (i.e. `exports.[“./types”].require`)
- (optional) An unformatted string with the error generated as-is today

With the first two points we’d already be able to pinpoint the issue to a specific line and tell a PR author that something is wrong & further inspection is needed.

I have put the third point as optional because the error/stack trace are kind of opaque (expected) so for us even just knowing there’s an issue with an export is already very valuable.

**Comment by webdeveric** [comment]
Thanks, that gives me enough to go on.

**Comment by webdeveric** [comment]
@dreamorosi I published version `0.2.0` which has the `--json` flag. Try it out and let me know if it covers your use case.

**Comment by dreamorosi** [comment]
Thank you for adding the feature, the format looks great!
